crates/bevy_math/src/primitives/mod.rs

```diff
@@ -13,6 +13,19 @@ pub trait Primitive2d {}
 /// A marker trait for 3D primitives
 pub trait Primitive3d {}
 
+/// An error indicating that a direction is invalid because it is zero or not finite.
+#[derive(Debug)]
+pub struct InvalidDirectionError;
```

Comment:
I think this is probably more useful as an enum: the different failure modes are useful to bubble up.

Reply:
I think we need to do a custom `try_normalize` for this, but I think it's easy since it's just an `.is_finite()` and a comparison against zero

Reply:
Or alternatively, do the checks again, but only in error cases

Reply:
I changed it to an enum with `Zero`, `Infinite` and `NaN` variants and added tests to verify that it gives the correct errors.